spring-ai-model/src/main/java/org/springframework/ai/tool/method/MethodToolCallback.java

```diff
@@ -125,11 +128,21 @@ private void validateToolContextSupport(@Nullable ToolContext toolContext) {
 	}
 
 	private Map<String, Object> extractToolArguments(String toolInput) {
-		return JsonParser.fromJson(toolInput, new TypeReference<>() {
-		});
+		try {
+			return JsonParser.fromJson(toolInput, new TypeReference<>() {
+			});
+		}
+		catch (IllegalStateException ex) {
```

Comment:
Instead of checking just the IllegalStateException, we agreed to check all exception which is similar to how MCP toolcallback error handling is done. I will make the change and update the PR.